### PR TITLE
[Children Ids] #184 - Create a cache to store children ids

### PR DESCRIPTION
**Pull request by PierreLeMaguer**
### Fixed Issues (if relevant)

[magento/community-features#183: Improve the performance of the method app/code/Magento/Eav/Model/Config.php::getAttribute](https://github.com/magento/community-features/issues/184)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=23910) <br/>All committers have signed the CLA.

File: app/code/Magento/Catalog/Model/Product/Type/AbstractType.php [hunk 1]
```diff
@@ -158,6 +158,11 @@ abstract class AbstractType
      */
     protected $_catalogProductOption;
 
+    /**
+     * @var array
+     */
+    protected $_cacheChildrenIds = [];
```
Comment (kandy) [inline app/code/Magento/Catalog/Model/Product/Type/AbstractType.php]: All new property must be private

**Comment by kandy** [comment]
PR is adding cache, but missed the cache cleanup that will cause issue

**Comment by sidolov** [comment]
@PierreLeMaguer , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Comment by romainruaud** [comment]
I'm reopening this PR (sorry if I'm exceeding my rights on this repository).

@sidolov @kandy can you give us clear directives about what needs to be changed / remain to be done ?

regards

**Review by sidolov — CHANGES_REQUESTED** [review]
@romainruaud , please, make `protected $_cacheChildrenIds = [];` as private, try to avoid underscores in property name. That's it

**Comment by PierreLeMaguer** [comment]
Hi @sidolov , the protected property is needed to use it in all child classes.

**Comment by sidolov** [comment]
@PierreLeMaguer , please, define private property in the each product type class instead of protected. Also, please, sign CLA, otherwise we cannot proceed with the pull request. Thank you.

**Comment by PierreLeMaguer** [comment]
Hi @sidolov , I changed the property for a private property.
Can you give us the next steps ?
Thanks

**Comment by sidolov** [comment]
@PierreLeMaguer please, take a look at failed tests, looks like the solution caused the behavior change

**Comment by sidolov** [comment]
@PierreLeMaguer  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**